src/storage.rs

```diff
@@ -0,0 +1,135 @@
+//! Storage
+/// Traits to allow on and off board storage deivces to read and write data
+/// Allows for Read Only Memory as well as Read and Write Memory
+use nb;
+
+/// Address represents an unsigned int. This allows for devices that have bigger or smaller address spaces than the host.
+pub struct Address<U>(pub U);
+/// Address Offset represents an unsigned int that is used as an optional offset from the base address.
+pub struct AddressOffset<U>(pub U);
+
+use core::ops::Add;
+
+/// Implement add for the Address and AddressOffset Types.
+impl<'a, 'b, U> Add<&'b AddressOffset<U>> for &'a Address<U>
```

Comment:
Subtraction of an offset from an address is probably also interesting. And if we already at it, maybe addition and subtraction of address offsets. However I start wondering where to stop :) and then if it makes sense to include these implementations at all since the user already needs to deal with the inner values.